lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -1070,17 +1067,14 @@ def mouse_init(self, rotate_btn=1, zoom_btn=3):
 
         """
         self.button_pressed = None
-        canv = self.figure.canvas
-        if canv is not None:
-            c1 = canv.mpl_connect('motion_notify_event', self._on_move)
-            c2 = canv.mpl_connect('button_press_event', self._button_press)
-            c3 = canv.mpl_connect('button_release_event', self._button_release)
-            self._cids = [c1, c2, c3]
-        else:
-            cbook._warn_external("Axes3D.figure.canvas is 'None', mouse "
-                                 "rotation disabled. Set canvas then call "
-                                 "Axes3D.mouse_init().")
-
+        self._cids = [
```

Comment:
Concerned that if we are hitting this code path with the base canvas and then it gets replaced the subscriptions will not work.

On the other hand, you are no worse than you are now.

Reply:
So I'll leave it as it is until someone complains :)